src/main/java/org/kuali/kra/subaward/service/impl/SubAwardServiceImpl.java

```diff
@@ -249,6 +251,17 @@ public SubAward getAmountInfo(SubAward subAward) {
         subAward.setTotalAmountReleased(totalAmountReleased);
         subAward.setTotalAvailableAmount(totalObligatedAmount.subtract(totalAmountReleased));
 
+        // BUKC-0145: Transactions on Financial Tab - Order switching (DFCT0011505)  - fixes issue with BUKC-0105
+        List<SubAwardAmountInfo> sortedlist = subAward.getSubAwardAmountInfoList();
+        if (!sortedlist.isEmpty() && !(sortedlist.get(sortedlist.size() - 1).getSubAwardAmountInfoId() == null)) {

```

Comment:
What is the purpose of the second condition: !(sortedlist.get(sortedlist.size() - 1).getSubAwardAmountInfoId() == null)? For the comparator to work, none of the subAwardAmountInfoId in the list can be null.... so I don't see why we test for just one of them....
Need to find out if it is even possible for them to be null and if that's the case, address it consistently in the comparator.